src/client/mod.rs

```diff
@@ -286,26 +338,12 @@ impl GaloyClient {
         Ok(())
     }
 
-    // TODO: check if we can do self without &
-    pub fn batch(self, filename: String, price: Decimal) -> anyhow::Result<()> {
-        let mut batch = Batch::new(self, price);
-
-        batch.add_csv(filename).context("can't load file")?;
-
-        batch
-            .populate_wallet_id()
-            .context("cant get wallet id for all username")?;
-
-        batch
-            .populate_sats()
-            .context("cant set sats all payments")?;
-
-        println!("going to execute:");
-        batch.show();
-
-        batch.execute().context("can't make payment successfully")?;
-
-        Ok(())
+    pub fn batch_payment(self, file: String) -> anyhow::Result<String> {
```

Comment:
Having batch as a submodule of client is IMO significantly bloating its responsibility. Typically a client is a thin wrapper around some network calls. More complex coordination that requires multiple round trips needs to be put in a use-case / app layer that uses the client as a dependency.